src/ts/component/block/chat.tsx

```diff
@@ -489,16 +514,62 @@ const BlockChat = observer(class BlockChat extends React.Component<I.BlockCompon
 		this.setState({ attachments: [] });
 	};
 
+	onScroll (e: any) {
+		const blockId = this.getBlockId();
+		const container = this.getScrollContainer();
+		const top = container.scrollTop() - $('#scrollWrapper').offset().top;
+		const form = $('#formWrapper');
+		const formPadding = Number(form.css('padding-bottom').replace('px', ''));
+		const viewport = container.outerHeight() - form.height() - formPadding;
+		const messages = this.getMessages();
+
+		const messagesIntoView = messages.filter((it) => {
+			const node = this.messagesMap[it.id].node;
+			const oT = node.offsetTop + node.clientHeight;
+
+			return oT >= top && oT < top + viewport;
+		});
+
+		if (!messagesIntoView.length) {
+			return;
+		};
+
+		const last = messagesIntoView[messagesIntoView.length - 1];
+		const lastNode = this.messagesMap[last.id].node;
+
+		if (lastNode.offsetTop > this.lastMessageOffset) {
+			this.lastMessageId = last.id;
+			this.lastMessageOffset = lastNode.offsetTop;
+			Storage.setLastChatMessageId(blockId, last.id);
+		};
+	};
+
+	scrollToMessage (id: string) {
+		window.setTimeout(() => {
```

Comment:
Why timeout?

Reply:
delay to let messages render correctly. moved it to the call